File: Prefix & Suffix Sum/1422. Maximum Score After Splitting a String.java
class Solution {
    public int maxScore(String s) {
        int n = s.length();
        int one = 0;

        for (int i = 0; i < n; i++) {
            if (s.charAt(i) == '1') {
                one++;
            }
        }

        int zero = 0;
        int max = 0;

        for (int i = 0; i < n - 1; i++) {
            if (s.charAt(i) == '1') {
                one--;
            } else {
                zero++;
            }

            max = Math.max(max, zero + one);
        }

        return max;
    }
}

File: LeetCode/Prefix & Suffix Sum Technique/1422_Maximum Score After Splitting a String.py
class Solution:
    def maxScore(self, s: str) -> int:
        ones = s.count('1')
        zeros = 0
        maxscore = 0
        for i in range(len(s) - 1):
            if s[i] == '0':
                zeros += 1
            else:
                ones -= 1
            maxscore = max(maxscore, zeros + ones)
        return maxscore

user_input = input("Enter a binary string: ")
solution = Solution()
output = solution.maxScore(user_input)
print(f"The maximum score for the given binary string is: {output}")
